src/sentry/integrations/vercel/webhook.py

```diff
@@ -142,27 +143,53 @@ def post(self, request: Request) -> Response:
 
         # Vercel's webhook allows you to subscribe to different events,
         # denoted by the `type` attribute. We currently subscribe to:
-        #     * integration-configuration-removed (Configuration Removed)
-        #     * deployment (Deployment Created)
+        #     * integration-configuration-removed, integration-configuration.removed (Configuration Removed)
+        #     * deployment, deployment.created (Deployment Created)
         # https://vercel.com/docs/integrations/webhooks-overview
-        try:
-            event_type = request.data["type"]
-        except KeyError:
-            return self.respond({"detail": "Missing event type."}, status=400)
+        with configure_scope() as scope:
+            try:
+                event_type = request.data["type"]
+            except KeyError:
+                return self.respond({"detail": "Missing event type."}, status=400)
+
+            # Try the new Vercel request. If it fails, try the old Vercel request
+            try:
+                payload = request.data["payload"]
+                external_id = (
+                    payload.get("team")["id"] if payload.get("team") else payload["user"]["id"]
+                )
+                scope.set_tag("vercel_webhook.type", "new")
 
-        external_id = request.data.get("teamId") or request.data["userId"]
+                if event_type == "integration-configuration.removed":
+                    configuration_id = request.data["payload"]["configuration"]["id"]
+                    return self._delete(external_id, configuration_id, request)
+                if event_type == "deployment.created":
+                    return self._deployment_created(external_id, request)
+            except KeyError:
+                external_id = request.data.get("teamId") or request.data["userId"]
+                scope.set_tag("vercel_webhook.type", "old")
 
-        if event_type == "integration-configuration-removed":
-            configuration_id = request.data["payload"]["configuration"]["id"]
-            return self._delete(external_id, configuration_id, request)
+                if event_type == "integration-configuration-removed":
+                    configuration_id = request.data["payload"]["configuration"]["id"]
+                    return self._delete(external_id, configuration_id, request)
 
-        if event_type == "deployment":
-            return self._deployment_created(external_id, request)
+                if event_type == "deployment":
+                    return self._deployment_created(external_id, request)
```

Comment:
Try the Vercel request with the new event type and payload first. If it fails, try the old one. Set a tag depending on which type it is.

All functionality remains the same, just the event type and how the payload is accessed is changed